numpy/_core/shape_base.py

```diff
@@ -233,9 +233,12 @@ def vstack(tup, *, dtype=None, casting="same_kind"):
 
     Parameters
     ----------
-    tup : sequence of ndarrays
+    tup : ndarray or sequence of ndarrays
         The arrays must have the same shape along all but the first axis.
-        1-D arrays must have the same length.
+        1-D arrays must have the same length. In the case of a single
+        array_like input, it will be treated a` a sequence of `shape[0]`
+        arrays, each of shape `shape[1:]`
+        (i.e. `[ary[i] for i in range(ary.shape[0])]`).
```

Comment:
What you have is probably clearer, but it is exactly the same as `[a for a in ary]`. Or perhaps not give the example? The single-array case really is the exception, so should not take more space than needed.